src/Development/IDE/Core/Rules.hs

```diff
@@ -553,16 +553,16 @@ getBindingsRule =
 getDocMapRule :: Rules ()
 getDocMapRule =
     define $ \GetDocMap file -> do
-      (tmrTypechecked -> tc,_) <- useWithStale_ TypeCheck file
-      (hscEnv -> hsc,_) <-useWithStale_ GhcSessionDeps file
-      (refMap -> rf, _) <- useWithStale_ GetHieAst file
+      (tmrTypechecked -> tc) <- use_ TypeCheck file
```

Comment:
I don't think `GetDocMap` is a core rule - it is only used for showing haddocks on hover, and I think it should be as resilient as possible.

The impact of it having the incorrect information is very low.

Reply:
Given that `getAtPoint` already calls `GetDocMap` with `useE`, can you produce an example where `useWithStale_` inside the `GetDocMap`rule will make it more resilient than `use_`?

Reply:
When we have a stale AST, but the user hasn't hovered, so we haven't computed a stale `DocMap` yet.

Reply:
Fair enough...